devutils/src/devutils/invoke_utils.py

```diff
@@ -197,15 +201,45 @@ def _make_conda_env_if_needed():
                 )
 
     @task
-    def install(ctx):
-        """Install dependencies"""
+    def install(ctx, local: str = None, update: bool = False, verbose: bool = False):
+        """
+        Installs or updates dependencies.
+
+        Args:
+            local: A comma-separated list of local projects to install in develop mode.
+            update: Whether to update the dependencies.
+            verbose: Whether to run in verbose mode.
+        """
         _make_conda_env_if_needed()
-        poetry(ctx, "install")
+
+        if update:
+            poetry(ctx, "update", verbose=verbose)
+            return
+
+        projects = local.split(",") if local else None
+        if projects:
+            with mark_as_develop_mode(projects):
+                poetry(ctx, "lock --no-update")
+                poetry(ctx, "install", verbose=verbose)
+        else:
+            poetry(ctx, "install", verbose=verbose)
 
     @task
-    def devinstall(ctx):
-        """Install dependencies with deps in dev mode"""
+    def devinstall(ctx, verbose: bool = False):
+        """
+        Install the package in develop mode and its dependencies.
+
+        Args:
+            verbose: Whether to run in verbose mode.
+        """
         _make_conda_env_if_needed()
+
+        with mark_as_develop_mode(all=True):
+            poetry(ctx, "lock --no-update")
+            poetry(ctx, "install", verbose=verbose)
+
+    @contextmanager
+    def mark_as_develop_mode(projects: Optional[list[str]] = None, all=False):
```

Comment:
```suggestion
    def mark_as_develop_mode(projects: Optional[list[str]] = None, all: bool = False):
```
if we still want to stay consistent with typing.